src/domotica/domotica_cv.c

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file domotica_cv.c
+ * @brief Process CV numbers for the loconet domotica module
+ *
+ * \copyright Copyright 2017 /Dev. All rights reserved.
+ * \license This project is released under MIT license.
+ *
+ * @author Jan Martijn van der Werf <janmartijn@slashdev.nl>
+ */
+
+#include "domotica_cv.h"
+
+// ----------------------------------------------------------------------------
+void loconet_cv_written_event(uint16_t lncv_number, uint16_t value)
+{
+  if (lncv_number == 0)
+  {
+    loconet_config.bit.ADDRESS = value;
+  }
+  else if (lncv_number == 2)
+  {
+    loconet_config.bit.PRIORITY = value;
+  }
+  else if (lncv_number == 3)
+  {
+    // Fast clock setting
+  }
+  else if (lncv_number >= DOMOTICA_LNCV_OUTPUT_BRIGHTNESS_START && lncv_number < DOMOTICA_LNCV_OUTPUT_BRIGHTNESS_END)
+  {
+    // Output light settings
+    if (value <= 0xFF)
+    {
+      domotica_set_output_brightness(lncv_number - DOMOTICA_LNCV_OUTPUT_BRIGHTNESS_START, (uint8_t) value);
+    }
+  }
+  else if (lncv_number >= DOMOTICA_LNCV_INPUT_ADDRESSES_START && lncv_number < DOMOTICA_LNCV_INPUT_ADDRESSES_END)
```

Comment:
Currently `DOMOTICA_LNCV_INPUT_ADDRESSES_START` is the same as `DOMOTICA_LNCV_OUTPUT_BRIGHTNESS_END`. When you would change the code that these can't be overridden, you won't have to check for `lncv_number >= DOMOTICA_LNCV_INPUT_ADDRESSES_START` anymore as the previous check already confirmed that `lncv_numer` is larger than `DOMOTICA_LNCV_OUTPUT_BRIGHTNESS_END`. Thus you'd only need to check if `lncv_number` is smaller than `DOMOTICA_LNCV_INPUT_ADDRESSES_END`.

The same goes for the if below for `DOMOTICA_LNCV_FASTCLOCK_START`.